### PR TITLE
More negative stride test coverage

### PR DESCRIPTION
**Pull request by dsharlet**

### CODE REVIEW

File: test/array.cpp [hunk 1]
```diff
@@ -409,13 +409,24 @@ TEST(array_reshape_3d) {
...
+  for_each_index(a_compact.shape(),
+      [&](const dense_array<int, 2>::index_type& i) { ASSERT_EQ(a_compact(i), std::get<1>(i)); });
+
+  // Test with a negative stride in x.
+  array_of_rank<int, 2> b({{0, 20, -1}, {0, 10}});
+  ASSERT_EQ(b.y().stride(), 20);
+  for_all_indices(b.shape(), [&](int x, int y) { b(x, y) = x; });
+
+  dense_array<int, 2> b_compact = make_compact_copy(b);
+  ASSERT(b_compact.x().stride() == 1);
```
Comment (jiawen) [inline test/array.cpp:427]: ASSERT_EQ?

File: test/array.cpp [hunk 2]
```diff
@@ -409,13 +409,24 @@ TEST(array_reshape_3d) {
...
+  // Test with a negative stride in y.
   array_of_rank<int, 2> a({{0, 10, 3}, {0, 3, -1}});
-  ASSERT_EQ(a.x().stride(), 3);
   for_all_indices(a.shape(), [&](int x, int y) { a(x, y) = y; });
 
-  dense_array<int, 2> b = make_compact_copy(a);
-  for_each_index(b.shape(),
-      [&](const dense_array<int, 2>::index_type& i) { ASSERT_EQ(b(i), std::get<1>(i)); });
+  dense_array<int, 2> a_compact = make_compact_copy(a);
+  ASSERT(a_compact.y().stride() > 0);
```
Comment (jiawen) [inline test/array.cpp:417]: ASSERT_GT

**Review by jiawen — APPROVED** [review]

File: test/array.cpp [hunk 3]
```diff
@@ -409,13 +409,24 @@ TEST(array_reshape_3d) {
 }
 
 TEST(array_negative_strides) {
+  // Test with a negative stride in y.
   array_of_rank<int, 2> a({{0, 10, 3}, {0, 3, -1}});
-  ASSERT_EQ(a.x().stride(), 3);
   for_all_indices(a.shape(), [&](int x, int y) { a(x, y) = y; });
```
Comment (jiawen) [inline test/array.cpp:414]: Say something like:

Initialize a(x, y) = y and check that it's still true if you make a compact copy that makes strides positive).

(and again for x).

File: test/array.cpp (hunk 2 above)
Comment (dsharlet) [reply test/array.cpp:417]: We only have ASSERT_LT :) This is just my local test.h...

File: test/array.cpp (hunk 3 above)
Comment (dsharlet) [reply test/array.cpp:414]: Rewrote this to use fill_pattern/check_pattern. I'm not sure why I didn't do that in the first place. Funny thing about codebases that only get a few commits every few months :)

**Pull request merged**